### ISSUE TITLE
NULL Pointer access when embed_documents

### ISSUE THREAD

**Issue by raymondshe** [abetlen/llama-cpp-python#1025]
When I was working on the samples here on my Mac M1 laptop. 

https://ai.plainenglish.io/%EF%B8%8F-langchain-streamlit-llama-bringing-conversational-ai-to-your-local-machine-a1736252b172

Everything looks fine until run "Part of the journey is the Embeddings !!!" section.

I was using:

embeddings = LlamaCppEmbeddings(model_path="/Users/raymond/moduls/mellogpt.Q4_K_M.gguf",n_gpu_layers= 1)

and the model load fine and I can use it for query. But when 

embedded_texts = embeddings.embed_documents(_texts)

The full code is at, but I changed some parameters shown above. 
https://miro.medium.com/v2/resize:fit:1400/format:webp/1*_Mw2-tecTljL8gF_IS6XGw.png

I got the problem:
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
ValueError: NULL pointer access

The above exception was the direct cause of the following exception:

SystemError                               Traceback (most recent call last)
Cell In[13], [line 1](vscode-notebook-cell:?execution_count=13&line=1)
----> [1](vscode-notebook-cell:?execution_count=13&line=1) embedded_texts = embeddings.embed_documents(_texts)
      [2](vscode-notebook-cell:?execution_count=13&line=2) len(embedded_texts), len(embedded_texts[0])

File [~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:113](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:113), in LlamaCppEmbeddings.embed_documents(self, texts)
    [104](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:104) def embed_documents(self, texts: List[str]) -> List[List[float]]:
    [105](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:105)     """Embed a list of documents using the Llama model.
    [106](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:106) 
    [107](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:107)     Args:
   (...)
    [111](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:111)         List of embeddings, one for each text.
    [112](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:112)     """
--> [113](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:113)     embeddings = [self.client.embed(text) for text in texts]
    [114](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:114)     return [list(map(float, e)) for e in embeddings]

File [~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:113](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:113), in <listcomp>(.0)
    [104](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:104) def embed_documents(self, texts: List[str]) -> List[List[float]]:
    [105](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:105)     """Embed a list of documents using the Llama model.
    [106](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:106) 
    [107](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:107)     Args:
   (...)
    [111](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:111)         List of embeddings, one for each text.
    [112](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:112)     """
--> [113](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:113)     embeddings = [self.client.embed(text) for text in texts]
    [114](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/langchain_community/embeddings/llamacpp.py:114)     return [list(map(float, e)) for e in embeddings]

File [~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1338](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1338), in Llama.embed(self, input)
   [1329](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1329) def embed(self, input: str) -> List[float]:
   [1330](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1330)     """Embed a string.
   [1331](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1331) 
   [1332](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1332)     Args:
   (...)
   [1336](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1336)         A list of embeddings
   [1337](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1337)     """
-> [1338](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1338)     return list(map(float, self.create_embedding(input)["data"][0]["embedding"]))

File [~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1309](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1309), in Llama.create_embedding(self, input, model)
   [1304](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1304)     total_tokens += n_tokens
   [1305](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1305)     embedding = llama_cpp.llama_get_embeddings(self._ctx.ctx)[
   [1306](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1306)         : llama_cpp.llama_n_embd(self._model.model)
   [1307](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1307)     ]
-> [1309](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1309)     data.append(
   [1310](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1310)         {
   [1311](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1311)             "object": "embedding",
   [1312](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1312)             "embedding": embedding,
   [1313](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1313)             "index": index,
   [1314](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1314)         }
   [1315](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1315)     )
   [1316](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1316) if self.verbose:
   [1317](https://file+.vscode-resource.vscode-cdn.net/Users/raymond/moduls/DocQA/~/miniforge3/envs/llama/lib/python3.9/site-packages/llama_cpp/llama.py:1317)     llama_cpp.llama_print_timings(self._ctx.ctx)

SystemError: <method 'append' of 'list' objects> returned a result with an error set

**Comment by raymondshe** [comment]
> python --version
Python 3.9.18
> pip show llama_cpp_python
Name: llama_cpp_python
Version: 0.2.23
Summary: Python bindings for the llama.cpp library
Home-page:
Author:
Author-email: Andrei Betlen <abetlen@gmail.com>
License: MIT
Location: /Users/raymond/miniforge3/envs/llama/lib/python3.9/site-packages
Requires: diskcache, numpy, typing-extensions
Required-by:

**Comment by brandonrobertz** [comment]
This should be fixed in latest with this PR: https://github.com/abetlen/llama-cpp-python/pull/1019

**Comment by abetlen** [comment]
Closing this as I believe @brandonrobertz PR fixed this issue, will reopen if issue still persists.

**Comment by raymondshe** [comment]
Thanks, I am going to try it.

**Comment by alexandrebcaruso** [comment]
still seems to be broken 
```
pip show llama_cpp_python
Name: llama_cpp_python
Version: 0.2.56
```
![image](https://github.com/abetlen/llama-cpp-python/assets/6711738/2d42d212-ac05-456a-9a41-682606db10f8)

![image](https://github.com/abetlen/llama-cpp-python/assets/6711738/36c8a7b0-79b3-4c6d-bcdf-0f8ac0368d85)

hope it helps

**Comment by brandonrobertz** [comment]
> still seems to be broken
> 
> ```
> pip show llama_cpp_python
> Name: llama_cpp_python
> Version: 0.2.56
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/6711738/314074597-2d42d212-ac05-456a-9a41-682606db10f8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA4NjU5ODMsIm5iZiI6MTcxMDg2NTY4MywicGF0aCI6Ii82NzExNzM4LzMxNDA3NDU5Ny0yZDQyZDIxMi1hYzA1LTQ1NmEtOWE0MS02ODI2MDZkYjEwZjgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMTlUMTYyODAzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjNkNWQxODEzMGM0Zjc5M2YyZjhhMjYzMDRmMTJhOTY0NzkxNWVjYTMyZjczNDQ4YjM2MjYwNWUxM2FiODg0MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.ab9VOVTkTIZrVVUxcFTgFo8_HgI2LbB6tjPfYJLUnUk)
> 
> ![image](https://private-user-images.githubusercontent.com/6711738/314074846-36c8a7b0-79b3-4c6d-bcdf-0f8ac0368d85.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA4NjU5ODMsIm5iZiI6MTcxMDg2NTY4MywicGF0aCI6Ii82NzExNzM4LzMxNDA3NDg0Ni0zNmM4YTdiMC03OWIzLTRjNmQtYmNkZi0wZjhhYzAzNjhkODUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMTlUMTYyODAzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDkxYTI2N2YzZTdmYmZhYjQ4NTYwNTMyYjk5NzNjYzUxMzFiOGI4YjYwMDBhMzcyMTg5YWMxZmI0MTlhZThiMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.pxJLrS5iNcBW5Q03h8W2OLbW_9Q2ajn5bULf3QdOdUs)
> 
> hope it helps

That's a different error. Open a new issue
